hack/image-mirroring/supplemental_ci_images_mirror_gen.py

```diff
@@ -0,0 +1,151 @@
+import os
+import sys
+import glob
+import shutil
+from ruamel.yaml import YAML
+
+APPCI_REGISTRY = "registry.ci.openshift.org"
+MAPPING_FILE_PREFIX = "mapping_"
+DIR_PATH = "core-services/image-mirroring"
+PERIODICS_FILE = "ci-operator/jobs/infra-image-mirroring-multi-arch.yaml"
+ARCHITECTURES = {
+    "arm64": "linux/arm64",
+}
+
+OWNERS = """approvers:
+  - dptp
+"""
+
+README = """This folder holds the mappings of the image mirroring for the multi architecture.
+
+Note that this folder is automatically generated.
+"""
+
+
+def generate_job(arch, os_filter):
+    return f"""
+agent: kubernetes
+cluster: app.ci
+cron: '@daily'
+decorate: true
+extra_refs:
+- base_ref: master
+  org: openshift
+  repo: release
+  workdir: true
+labels:
+  ci.openshift.io/area: supplemental-ci-images-{arch}
+  ci.openshift.io/role: infra
+name: periodic-image-mirroring-supplemental-ci-images-{arch}
+spec:
+  containers:
+  - args:
+    - --skip-missing=true
+    - --filter-by-os={os_filter}
+    - --skip-multiple-scopes
+    command:
+    - hack/image-mirroring/mirror-images.sh
+    env:
+    - name: ARCH
+      value: {arch}
+    - name: HOME
+      value: /home/mirror
+    image: registry.ci.openshift.org/ocp/4.12:cli
+    imagePullPolicy: Always
+    name: ""
+    resources:
+      requests:
+        cpu: 500m
+    volumeMounts:
+    - mountPath: /home/mirror/.docker/config.json
+      name: push
+      readOnly: true
+      subPath: .dockerconfigjson
+  volumes:
+  - name: push
+    secret:
+      secretName: registry-push-credentials-ci-central
+"""
+
+
+def generate_mappings():
+    print("Removing all core-services/image-mirroring-* directories")
+    for path in glob.glob('core-services/image-mirroring-*', recursive=False):
+        try:
+            shutil.rmtree(path, ignore_errors=False, onerror=None)
+        except OSError as error:
+            print(f"Error while deleting file {path}: {error}")
+
+    for root, dirs, files in os.walk(DIR_PATH):
+        if len(dirs) > 0:
+            continue
+
+        for name in files:
+            if not name.startswith(MAPPING_FILE_PREFIX):
+                continue
+            filename = f"{root}/{name}"
+
+            with open(filename, 'r', encoding="utf-8") as file:
+                lines = file.readlines()
+                for line in lines:
+                    line = line.strip()
+                    if len(line.strip()) == 0:
+                        continue
+                    # Ignore commented lines or if the mapping source is registry.ci.openshift.org
+                    # to avoid mirroring from app.ci to an external source
+                    if line.startswith("#") or line.startswith(APPCI_REGISTRY):
+                        continue
+
+                    source, dest = line.split()
+
+                    if not dest.startswith(APPCI_REGISTRY):
+                        print("BUG: where are we mirroring to?")
```

Comment:
Could this comment be a little more descriptive?

Reply:
I added a comment!